tests/Operation/FindTest.php

```diff
@@ -154,4 +156,61 @@ public function provideInvalidConstructorCursorTypeOptions()
     {
         return $this->wrapValuesForDataProvider([-1, 0, 4]);
     }
+
+    public function testExplainableCommandDocument(): void
+    {
+        // all options except deprecated "snapshot" and "maxScan"
+        $options = [
+            'allowDiskUse' => true,
+            'allowPartialResults' => true,
+            'batchSize' => 123,
+            'collation' => ['locale' => 'fr'],
+            'comment' => 'explain me',
+            'cursorType' => Find::NON_TAILABLE,
+            'hint' => '_id_',
+            'limit' => 15,
+            'max' => ['x' => 100],
+            'maxAwaitTimeMS' => 500,
+            'maxTimeMS' => 100,
+            'min' => ['x' => 10],
+            'modifiers' => ['foo' => 'bar'],
+            'noCursorTimeout' => true,
+            'oplogReplay' => true,
+            'projection' => ['_id' => 0],
+            'readConcern' => new ReadConcern(ReadConcern::LOCAL),
+            'readPreference' => new ReadPreference(ReadPreference::SECONDARY_PREFERRED),
+            'returnKey' => true,
+            'showRecordId' => true,
+            'skip' => 5,
+            'sort' => ['x' => 1],
+            'let' => ['y' => 2],
+            'typeMap' => ['root' => 'array'],
+        ];
+        $operation = new Find($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], $options);
+
+        $expected = [
+            'find' => $this->getCollectionName(),
+            'filter' => (object) ['x' => 1],
+            'allowDiskUse' => true,
+            'allowPartialResults' => true,
+            'batchSize' => 123,
+            'comment' => 'explain me',
+            'hint' => '_id_',
+            'limit' => 15,
+            'maxTimeMS' => 100,
+            'noCursorTimeout' => true,
+            'oplogReplay' => true,
+            'projection' => ['_id' => 0],
+            'readConcern' => new ReadConcern(ReadConcern::LOCAL),
```

Comment:
I suppose it's not worth adding a `@todo` comment here since the code won't need to be changed in PHPLIB. This is worth calling out in the commit message, though (with a reference to PHPC-2234).